### DISCUSSION TITLE
How can I play with cohttp-eio now?

### DISCUSSION THREAD

**Discussion by dangdennis** [mirage/ocaml-cohttp#944] [Q&A] [answered]
Since there’s no opam package yet for cohttp on the ocaml 5.0 beta switch, can you recommend a path to try out cohttp with eio?

Ive gone as far as just vendoring http, cohttp, and cohttp-eio locally just to run it. Is there a simpler way that lets me continually pull any dev changes?

**Comment by mseri** [comment, marked as answer]
If you use opam, you can clone this repository and pin it wil `opam pin . —-with-version 6.0.0~alpha0`. I was planning to release an alpha soon, I will try to get to it next week.

**Reply by dangdennis** [reply]
Ooh thank you. I’ll try this asap.

**Reply by mseri** [reply]
It should work if you do something like `opam pin git@github.com:mirage/ocaml-cohttp —with-versiom 6.0.0~alpha0`, in that way you don’t need to manually pull

**Reply by dangdennis** [reply]
`opam pin add https://github.com/mirage/ocaml-cohttp.git --with-version 6.0.0~alpha0` is the working command now.

i couldn't get `git@...` to work. iirc, github disabled download of git@urls from non-ssh origins.

the error.
```
[ERROR] Could not synchronize
        /private/var/folders/33/dxhswk150sx1sblqvj718r700000gn/T/opam-pin-cache-40767-40931e/1ba8a617f21aaf26
        from "git+ssh://git@github.com/mirage/ocaml-cohttp":
        "/opt/homebrew/bin/git fetch -q" exited with code 128
[ERROR] Could not retrieve git+ssh://git@github.com/mirage/ocaml-cohttp
```

cohttp-eio works! thanks for the work @mseri @bikallem

**Reply by mseri** [reply]
If you pin the https url it works?

**Reply by dangdennis** [reply]
It did work.
